### ISSUE TITLE
Refresh on startup

### ISSUE THREAD

**Issue by Donkey-Doug** [WangDaYeeeeee/GeometricWeather#378]
I need to manually refresh every time I open the app.

**Comment by papjul** [comment]
Do you use a customized Android OS?
Did you allow to run in background (without being killed)?

**Comment by Donkey-Doug** [comment]
I use lineageos.

I added a manual location.

It is allowed to run in the background and it does update at the specified interval in the background. But if my interval is set to 30 min and 12 min have past, then my info is 18 min outdated when I open the app. So updating in the background does work.

It does not update on startup, nor is there an option to update on startup in settings. Therefore I assumed update on startup is not implemented.

**Comment by papjul** [comment]
Alright, I thought you reported a bug, not a new feature ;)

However, if you open the app every minute, I don't think the app should trouble the weather server everytime. Plus, that would add workload that would slowdown your interaction with the app everytime you open it. We have many reports that the app is already slow on startup so I really don't think that would be a good idea to allow such a thing.

**Comment by Donkey-Doug** [comment]
Then make it optional. A toggle in settings to update at startup.

**Comment by papjul** [comment]
We would need #244 first, as this kind of behavior is contrary to terms of service, and would uselessly consume calls.

**Comment by papjul** [comment]
Actually, I just tested it and what you asked for is already implemented.

Geometric Weather will refresh weather data on startup if background task failed to do it within the configured time in settings.

30 minutes is the minimum allowed in Geometric Weather. If you want shorter options, then your issue is a duplicate of #322.

**Comment by netsonicyxf** [comment]
I have the same issue on my Samsung official oneui4.1, android 12.
I disabled “后台纯净”，and set 1hr as update interval, but it never auto updates.

**Comment by glocalglocal** [comment]
> Geometric Weather will refresh weather data on startup if background task failed to do it within the configured time in settings.

In my case, checking on startup would result in fewer connections actually. I would prefer the app not to check or run in the background at all. Instead, I would want it to refresh only the moment I open it, which may be once every few days, saving battery and calls. I am not interested in alerts anyway. 

To prevent unnecessary calls when the user reopens the app within minutes, the startup call could be skipped if the data are, say, less than 30 minutes old.

**Comment by papjul** [comment]
> I would want it to refresh only the moment I open it, which may be once every few days, saving battery and calls. I am not interested in alerts anyway.
> 
> To prevent unnecessary calls when the user reopens the app within minutes, the startup call could be skipped if the data are, say, less than 30 minutes old.

Yes, this is how it works:

> Geometric Weather will refresh weather data on startup if background task failed to do it within the configured time in settings.

You could kill app and let background optimization enabled as a workaround for the "no refresh in the background" part.

**Comment by glocalglocal** [comment]
> You could kill app and let background optimization enabled as a workaround for the "no refresh in the background" part.

I was suggesting a more elegant way to do this: (a) adding a 'no background refresh' option to the existing list of frequencies, (b) then the app would automatically stop running in the background as it can't alert the user about something new anyway (c) refreshing on startup but only if data are stale (eg over 30 minutes old). 

That results in fewer refreshes and removes the need for manual refresh.

**Comment by papjul** [comment]
I don't think it would be compatible with an automatic refresh on startup unless we add an additional setting. Because how do you decide if the user wants 30 min, 1 h, 4 h, etc ?

**Comment by glocalglocal** [comment]
@papjul an additional setting is a possibility, but I would be in favour of making the harmless assumption that users who select 'no background refresh' will more likely have stale data most of the times, because that's what they probably want: rare refreshes on demand. For startup-only refreshes 30 mins is already considered safe and could be the default. 

In other words, unless you want alerts, why would you want an app to refresh in the background at all? Some people like myself don't even use widgets. I would expect it to go well away and stay there until I run it. I appreciate apps that stay out of the way and leave my resources alone.
